crates/data_components/src/dynamodb/provider.rs

```
@@ -237,44 +182,63 @@ impl TableProvider for DynamoDBTableProvider {
         &self,
         _state: &dyn Session,
         projection: Option<&Vec<usize>>,
-        _filters: &[Expr],
+        filters: &[Expr],
         limit: Option<usize>,
-    ) -> datafusion::error::Result<Arc<dyn ExecutionPlan>> {
-        let mut request = self.client.scan().table_name(self.table_name.to_string());
-        if let Some(limit) = limit {
-            request = request.limit(
-                i32::try_from(limit)
-                    .map_err(|_| DataFusionError::Execution("Limit is too large".to_string()))?,
-            );
-        }
-        if let Some((projection, names)) = projection_expression(projection, &self.table_schema) {
-            request = request.projection_expression(projection);
-            for (placeholder, name) in &names {
-                request = request.expression_attribute_names(placeholder, name);
-            }
+    ) -> DataFusionResult<Arc<dyn ExecutionPlan>> {
+        let mut projected_schema = project_schema(self.table_schema.schema(), projection)?;
+
+        // If no columns are specified, use partition_key - otherwise DynamoDB returns an error
+        if projected_schema.fields.is_empty() {
```

This will break the schema comparison when this is executed, we should just return an error at this point.

---

This is needed in cases like `select *` - datafusion passes empty projection. There are integration tests that verify this behavior

---

won't that break the schema comparison though?